aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -218,13 +221,32 @@ def __init__(
         cors: bool,
         compress: bool,
         cache_control: Optional[str],
+        middleware: Optional[List[Callable]],
     ):
         self.method = method.upper()
         self.rule = rule
         self.func = func
         self.cors = cors
         self.compress = compress
         self.cache_control = cache_control
+        self.middleware = middleware or []
+
+    def __call__(self, router_middleware: List[Callable], app, args: Dict[str, str]):
```

Comment:
```suggestion
    def __call__(self, router_middlewares: List[Callable], app, route_arguments: Dict[str, str]):
```

Old tech debt we never paid. `args` actually meant options for dynamic routes that were resolved 

Given a registered route as `/account/<account_id>`, `args` would then become: `{"account_id": "value"}`.

Could you update the other parts too given it's private methods?

Tks!!!!!

https://github.com/aws-powertools/powertools-lambda-python/blob/4ff9ff17922d27a967c8946e98873d4e35d97ca3/aws_lambda_powertools/event_handler/api_gateway.py#L705